tfhe/src/integer/server_key/radix_parallel/mul.rs

```diff
@@ -755,4 +756,106 @@ impl ServerKey {
 
         self.unchecked_unsigned_overflowing_mul_assign_parallelized(lhs, rhs)
     }
+
+    /// Computes homomorphically a multiplication along with an overflow flag
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// // Generate the client key and the server key:
+    /// let num_blocks = 4;
+    /// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, num_blocks);
+    ///
+    /// let clear_1 = -128i8;
+    /// let clear_2 = 5i8;
+    ///
+    /// // Encrypt two messages
+    /// let ctxt_1 = cks.encrypt_signed(clear_1);
+    /// let ctxt_2 = cks.encrypt_signed(clear_2);
+    ///
+    /// // Compute homomorphically a multiplication
+    /// let (ct_res, ct_overflowed) = sks.signed_overflowing_mul_parallelized(&ctxt_1, &ctxt_2);
+    /// // Decrypt
+    /// let res: i8 = cks.decrypt_signed(&ct_res);
+    /// let overflowed = cks.decrypt_bool(&ct_overflowed);
+    ///
+    /// let (expected_result, expected_overflowed) = clear_1.overflowing_mul(clear_2);
+    /// assert_eq!(res, expected_result);
+    /// assert_eq!(overflowed, expected_overflowed);
+    /// ```
+    pub fn signed_overflowing_mul_parallelized(
+        &self,
+        lhs: &SignedRadixCiphertext,
+        rhs: &SignedRadixCiphertext,
+    ) -> (SignedRadixCiphertext, BooleanBlock) {
+        // Note: Naive implementation of signed mul with overflow
+        // surely there are optimized way of computing this
+
+        // Do a full multiplication (Nbits * Nbits) = 2Nbits result
+        let bigger_lhs = self.extend_radix_with_sign_msb(lhs, lhs.blocks.len());
+        let bigger_rhs = self.extend_radix_with_sign_msb(rhs, rhs.blocks.len());
+        let mut full_result = self.mul_parallelized(&bigger_lhs, &bigger_rhs);
+
+        let num_bits_of_message = self.message_modulus().0.ilog2();
+        let total_num_bits = num_bits_of_message * lhs.blocks.len() as u32;
+
+        let (is_lt_min, is_gt_max) = if total_num_bits > I256::BITS {
+            // The max of a two's complement number is a 0 in msb and then only 1
+            // The min is the opposite, 1 in msb, then zero
```

Comment:
do we have a test for the 256 bit case ?

follow up question : is there a way to have a dynamic bigint that would be helpful here ? Not necessary for the PR and not sure it's something we really need at the moment

Reply:
We don't have dynamic bigint but we have all the backbone stuff already implemented for it

I tested this path localy, maybe I can add a small case in the test to do 1 or 2 mul > 256 bits

Reply:
Or slightly better maybe, put the code that builds the trivial min/max into separate  pub functions, that we test

Reply:
if you feel that's better and it does not take you too much time why not yes